### PR TITLE
Add Non-squeamish trait

### PR DESCRIPTION
**Pull request by Night-Pryanik**
Non-squeamish characters consider filthy items as normal (no `(filthy)` suffix, no brown color etc) and can wear them without morale penalties.
Also, changed amount of water used to wash items from fixed 40 charges to proportional to item's volume. 
I wanted to change amount of time used to wash items too, but I couldn't find a good enough formula for it.

### CODE REVIEW

**Comment by codemime** [comment]
Please add a consistency check for `nonsqueamish` to `player_morale::consistent_with()` (same as for `took_prozac`, `stylish` e.t.c).

File: data/json/mutations.json [hunk 1]
```diff
@@ -4867,6 +4867,15 @@
...
+        "description" : "You can't even think about putting filthy clothes on yourself, especially from zombies' corpses.",
+        "cancels" : ["NONSQUEAMISH"]
+    },{
+        "type" : "mutation",
+        "id" : "NONSQUEAMISH",
+        "name" : "Non-squeamish",
+        "points" : 1,
+        "starting_trait" : true,
+        "description" : "You just don't care much about being dirty.  You consider filthy items as normal.",
+		"cancels" : ["SQUEAMISH"]
```
Comment (codemime) [inline data/json/mutations.json]: Tabs.

File: src/item.cpp [hunk 2]
```diff
@@ -6015,5 +6015,5 @@ bool item_category::operator!=( const item_category &rhs ) const
 }
 
 bool item::is_filthy() const {
-    return has_flag( "FILTHY" );
+    return has_flag( "FILTHY" ) && !g->u.has_trait( "NONSQUEAMISH" );
```
Comment (codemime) [inline src/item.cpp]: The function should probably be renamed back to `is_disgusting_for()` and accept an argument (just like it did earlier).

File: src/iuse.cpp [hunk 3]
```diff
@@ -8090,22 +8090,22 @@ int iuse::washclothes( player *p, item *it, bool, const tripoint& )
...
     const int pos = g->inv_for_flag( "FILTHY", _( "Wash what?" ) );
     item &mod = p->i_at( pos );
     if( pos == INT_MIN ) {
         p->add_msg_if_player( m_info, _( "Never mind." ) );
         return 0;
     }
-
+    
+    const inventory &crafting_inv = p->crafting_inventory();
+    if( !crafting_inv.has_charges( "water", 3 * mod.volume() ) && !crafting_inv.has_charges( "water_clean", 3 * mod.volume() ) ) {
```
Comment (codemime) [inline src/iuse.cpp]: `3 * mod.volume()` occurs four times. I think a constant should be used here:
`const auto required_water_charges  = 3 * mod.volume()`.

File: src/iuse.cpp [hunk 4]
```diff
@@ -8090,22 +8090,22 @@ int iuse::washclothes( player *p, item *it, bool, const tripoint& )
...
+        p->add_msg_if_player( _( "You need amount of water equal to thrice the volume of the item you want to wash." ) );
```
Comment (codemime) [inline src/iuse.cpp]: You need **an** amount...

File: src/morale.cpp [hunk 5]
```diff
@@ -781,22 +789,33 @@ void player_morale::update_constrained_penalty()
...
-                bp_pen( bp_leg_l,  5 ) +
-                bp_pen( bp_leg_r,  5 ) +
-                bp_pen( bp_arm_l,  5 ) +
-                bp_pen( bp_arm_r,  5 ) +
-                bp_pen( bp_foot_l, 3 ) +
-                bp_pen( bp_foot_r, 3 ) +
-                bp_pen( bp_hand_l, 3 ) +
-                bp_pen( bp_hand_r, 3 ) );
+
+    if( !nonsqueamish ) {
```
Comment (codemime) [inline src/morale.cpp]: Double negations look weird. I'd rename the variable back to `squeamish`.

Comment (Night-Pryanik) [inline src/morale.cpp]: Well, by negating `nonsqueamish` I meant squeamish OR ordinary person.

Comment (codemime) [inline src/morale.cpp]: It's just a matter of naming: `squeamish == !nonsqueamish`.

Comment (Night-Pryanik) [inline src/morale.cpp]: In terms of pseudocode I view it like this:
`!nonsqueamish == squeamish /**(strong squeamish)*/ || ordinary /**(soft squeamish)*/`

**Comment by Coolthulhu** [comment]
If filthiness is to apply morale penalty, it should be off by default, not with a trait.
Morale penalty for worn equipment is not a good idea.

**Comment by Night-Pryanik** [comment]
What do you propose? Remove this new non-squeamish trait and remove filthiness morale penalty for ordinary characters, leaving it only for squeamish characters?

**Comment by Coolthulhu** [comment]
Yes.

Alternatively rework squeamishness to avoid the design problems with morale penalty being tied to armor values.

The problem with squeamishness is pure tedium - morale penalty affects crafting, armor boosts combat. You can safely take off all the filthy gear before crafting/reading, it will only take player time, not character time. This makes optimal gameplay tedious, but doesn't add any difficulty.

The original incarnation of squeamish trait - never being able to wear zombie-dropped armor - was OK from design point of view, because it meant being unable to access some rare gear without lucking out and finding a clean spawn of it.
Current one is horrible - it doesn't prohibit wearing rare gear (defeating the point), it only requires taking it to a water source once and having some soap.
The non-trait version is even worse - it doesn't cause any noticeable penalties if the player takes the tedium to take it off. Rewarding the tedium is an inherently bad idea.

**Comment by Night-Pryanik** [comment]
Accepting your point of view, I could call "tedious" a lot of things currently implemented in game.
The necessity for eating/drinking first comes to mind. There are a lot of food and drinks in game. Finding some food and drinks is not a big deal, and maintaining hunger and thirst 

>will only take player time, not character time. This makes optimal gameplay tedious, but doesn't add any difficulty.

This is only one example, and I can give some more, but I think the message is clear.

**Comment by Coolthulhu** [comment]
> The necessity for eating/drinking first comes to mind.

Necessity of eating and drinking wouldn't be solved by automatic button pressing.
You still need to find food.

When designing, it's a good idea to consider a bot making automatic decisions vs. a human having to press buttons. If the bot's situation looks better, something is wrong.
In this case, the bot would just take off all filthy clothing and put on a blanket before crafting, thus avoiding all the filthiness penalties during downtime - crafting, sleeping, working on a car.

**Comment by Night-Pryanik** [comment]
> just take off all filthy clothing

And what about situation in which you can't do that? If the season is winter, for example?

**Comment by Coolthulhu** [comment]
> And what about situation in which you can't do that?

Put on a nearby blanket/sheet/a bunch of easy to craft but encumbering clothing made from rags.

**Comment by codemime** [comment]
> And what about situation in which you can't do that? If the season is winter, for example?

I have and idea which could mitigate the design flaw pointed by @Coolthulhu. What if one can't get rid of the penalty by simply taking off their clothes: the skin is already soiled and the filth has to be washed away?

**Comment by Night-Pryanik** [comment]
> Put on a nearby blanket/sheet/a bunch of easy to craft but encumbering clothing made from rags.

Rephrasing you, you still need to find/craft them.

I start to think that it's a senseless dispute. We can't convince each other. I don't think it's a design flaw or tedium at all. I think the system adds interest, difficulty and realism - for me, at least.

**Comment by Coolthulhu** [comment]
> the skin is already soiled and the filth has to be washed away

An improvement, but one that adds two possible problems:
* If filthiness is to be permanent until removed, that's a significant penalty for wearing filthy clothing that may make it far too punishing to be considered.
* If it is to be temporary but shorter than duration of sleep, it only partially mitigates the issue

> I start to think that it's a senseless dispute. We can't convince each other.

I'm open to arguments, not "for me it kinda works" (ie. opinions).
Filthiness has a significant cost in keypresses, but doesn't have one in in-game resources if you do the keypresses.

> you still need to find/craft them.

Which is trivial vast majority of the time due to how common rags are. Defeating the point of filthiness vast majority of the time.

**Comment by codemime** [comment]
> If filthiness is to be permanent until removed, that's a significant penalty for wearing filthy clothing 

Then wearing them could become more rewarding. For example, for mitigating player's scent / making [the player less visible and attractive for zombies](http://www.eonline.com/eol_images/Entire_Site/20151030/rs_1024x759-151130120708-1024.the-walking-dead-ponchos-2.ch.113015.jpg). It's a fair trade off: suffering for improved safety.

**Comment by Night-Pryanik** [comment]
I don't want to argue anymore.
First - this PR is not suitable place for this.
Second - my arguments are all based on my opinion. 

If you don't like the system, feel free to revert it back to initial state, or even remove it completely.

**Pull request closed without merging**